handler/src/main/java/io/netty/handler/ssl/AbstractSniHandler.java

```
@@ -0,0 +1,323 @@
+                                        offset++;
+
+                                        if (serverNameType == 0) {
+                                            final int serverNameLength = in.getUnsignedShort(offset);
+                                            offset += 2;
+
+                                            if (extensionsLimit - offset < serverNameLength) {
+                                                break loop;
+                                            }
+
+                                            final String hostname = in.toString(offset, serverNameLength,
+                                                                                CharsetUtil.UTF_8);
+
+                                            try {
+                                                select(ctx, IDN.toASCII(hostname,
+                                                                        IDN.ALLOW_UNASSIGNED).toLowerCase(Locale.US));
+                                            } catch (Throwable t) {
+                                                PlatformDependent.throwException(t);
+                                            }
+                                            return;
+                                        } else {
+                                            // invalid enum value
+                                            break loop;
+                                        }
+                                    }
+
+                                    offset += extensionLength;
+                                }
+                            }
+                            // Fall-through
+                        default:
+                            //not tls, ssl or application data, do not try sni
+                            break loop;
+                    }
+                }
+            } catch (Throwable e) {
+                // unexpected encoding, ignore sni and use default
```

only encoding? Why this swallowing any Throwable?